pkg/ip/ip.go

```diff
@@ -806,3 +806,71 @@ func IPToPrefix(ip net.IP) *net.IPNet {
 func IsIPv4(ip net.IP) bool {
 	return ip.To4() != nil
 }
+
+// IsIPv6 returns if netIP is IPv6.
+func IsIPv6(ip net.IP) bool {
+	return ip != nil && ip.To4() == nil
+}
+
+// SortIPList sorts the provided net.IP slice in place.
+func SortIPList(ipList []net.IP) {
+	sort.Slice(ipList, func(i, j int) bool {
+		return bytes.Compare(ipList[i], ipList[j]) < 0
+	})
+}
+
+// getSortedIPList returns a new net.IP slice in which the IPs are sorted.
+func getSortedIPList(ipList []net.IP) []net.IP {
+	sortedIPList := make([]net.IP, len(ipList))
+	for i := 0; i < len(ipList); i++ {
+		sortedIPList[i] = ipList[i]
+	}
+
+	SortIPList(sortedIPList)
+	return sortedIPList
+}
+
+// SortedIPListsAreEqual compares two lists of sorted IPs. If any differ it returns
+// false.
+func SortedIPListsAreEqual(a, b []net.IP) bool {
+	// The IP set is definitely different if the lengths are different.
+	if len(a) != len(b) {
+		return false
+	}
+
+	// Lengths are equal, so each member in one set must be in the other
+	// If any IPs at the same index differ the sorted IP list are not equal.
+	for i := range a {
+		if !a[i].Equal(b[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+// UnsortedIPListsAreEqual returns true if the list of net.IP provided is same
+// without considering the order of the IPs in the list. The function will first
+// attempt to sort both the IP lists and then validate equality for sorted lists.
+func UnsortedIPListsAreEqual(ipList1, ipList2 []net.IP) bool {
```

Comment:
I see you've implemented both variants for checking IP list equality, sorted and unsorted. The latter will simply sort the lists anyway. I guess the assumption being made here is that the caller should know if the IP lists they're passing into these functions are either sorted and unsorted. I'm curious as to why you chose to leave it up to the caller? Should they know if their IP lists are (un)sorted?

Seems to me that if we already sort the lists, then we might as well just have `AreIPListsEqual()` where we sort them under-the-hood, without requiring the caller to know if they are already sorted or not. WDYT?

Reply:
[General comments, no idea how callers of this work]

The issue with sorting the lists under the hood is that previously kept indices will become invalid. I would prefer keeping the code as is. If it is decided to sort things under the hood, please at least rename  `AreIPListsEqual()` to something that indicates that the order might change.

Reply:
That's a good point ^

Reply:
I agree with @kkourt 
The idea was that it is the responsibility of the caller to know what kind of equality should be checked for.